services/py-api/src/server/handlers/participants_handlers.py

```diff
@@ -21,14 +25,27 @@ def __init__(self, service: ParticipantRegistrationService) -> None:
         self._service = service
 
     async def create_participant(
-        self, response: Response, input_data: ParticipantRequestBody
+        self, response: Response, input_data: ParticipantRequestBody, jwt_token: Optional[str] = None,
     ) -> ParticipantRegisteredResponse | ErrResponse:
+        
         if input_data.is_admin and input_data.team_name:
             result = await self._service.register_admin_participant(input_data)
 
-        # TODO: Implement this when the invite participant case is done
-        # elif input_data.is_admin is False and input_data.team_name:
-        #     result = await self._service.register_invite_participant(input_data)
+        elif input_data.is_admin is False and input_data.team_name:
+            # Decode the token
+            decoded_result = JwtUtility.decode_data(token=jwt_token, schema=JwtUserData)
+            if is_err(decoded_result):
+                response.status_code = status.HTTP_401_UNAUTHORIZED
+                return ErrResponse(error=f"JWT decoding failed: {decoded_result.err_value}")
+            
+            decoded_data = decoded_result.ok_value
+            team_name_from_token = decoded_data["team_id"]
```

Comment:
team_name is not the same as team_id. The the name property of a team is a string while the _id property of a team is an Object Id, so you will need to change this check.